app/src/main/java/com/worldsnas/starplayer/ExoPlayerService.kt

```diff
@@ -0,0 +1,425 @@
+            }
+            val pendingIntentNext = PendingIntent.getService(
+                this,
+                NOTIFICATION_ID.FOREGROUND_SERVICE,
+                generateNextIntent(this),
+                PendingIntent.FLAG_UPDATE_CURRENT
+            )
+            notificationBuilder.addAction(
+                NotificationCompat.Action.Builder(
+                    R.drawable.exo_notification_next,
+                    "بعدی",
+                    pendingIntentNext
+                ).build()
+            )
+            val pendingIntentStop = PendingIntent.getService(
+                this,
+                NOTIFICATION_ID.FOREGROUND_SERVICE,
+                generateStopIntent(this),
+                PendingIntent.FLAG_UPDATE_CURRENT
+            )
+            notificationBuilder.addAction(
+                NotificationCompat.Action.Builder(
+                    R.drawable.exo_notification_stop,
+                    "بستن",
+                    pendingIntentStop
+                ).build()
+            )
+            startForeground(
+                NOTIFICATION_ID.FOREGROUND_SERVICE,
+                notificationBuilder.build()
+            )
+        }
+        return music
+    }
+
+    private fun releasePlayer() {
+//        if (player != null) {
+        playbackPosition = player.currentPosition
+        currentWindow = player.currentWindowIndex
+        player.stop()
+        player.release()
+//            player = null
+//        }
+    }
+
+    override fun onBind(intent: Intent): IBinder? {
+        return null
+    }
+
+    override fun onLoadingChanged(isLoading: Boolean) {}
+    override fun onPlayerStateChanged(
+        playWhenReady: Boolean,
+        playbackState: Int
+    ) {
+        when (playbackState) {
+            Player.STATE_BUFFERING -> {
+            }
+            Player.STATE_ENDED -> {
+                player.seekTo(
+                    player.currentWindowIndex,
+                    0
+                )
+                pausePlayer()
+            }
+
+            Player.STATE_READY -> {
+               if (!playWhenReady)
+                   showCurrentTrackNotification(musicList?.get(player.currentWindowIndex))
+            }
+
+        }
+
+    }
+
+    override fun onTracksChanged(
+        trackGroups: TrackGroupArray,
+        trackSelections: TrackSelectionArray
+    ) {
+        super.onTracksChanged(trackGroups, trackSelections)
+        showCurrentTrackNotification(musicList?.get(player.currentWindowIndex))
+    }
+
+    override fun onPlayerError(error: ExoPlaybackException) {
+        pausePlayer()
+        player.prepare(mediaSource!!)
+        player.seekTo(currentWindow, playbackPosition)
+//        broadCastCurrentTrack()
+//        sendBroadcast_stateIdle()
+    }
+
+    override fun onPositionDiscontinuity(reason: Int) {
+
+    }
+
+    override fun onPlaybackParametersChanged(playbackParameters: PlaybackParameters) {}
+    override fun onSeekProcessed() {}
+    interface ACTION {
```

Comment:
chera interface sakhti?
age companion object mikahy ke esm dashte bashe intori ham mishe:
`companion object Action`
albate benazaram aslan niazi be companion nist

Reply:
ok

Reply:
dorostesh mikonam